www/content/workstation/upgrade_labs.md

```diff
@@ -0,0 +1,454 @@
+
+The Upgrade Lab assumes--but does not require--that you have a continuous integration pipeline (CI) setup for your cookbooks with:
+
+  * A version control system (for example, git)
+  * Some degree of automated testing for proposed changes
+  * A continuous delivery system (CD) that controls cookbook releases; the CD is the mechanism for updating cookbook versions and uploading them to the Chef Server(s)
+
+If you do not have a version control system and CI/CD pipeline in place, then please contact [Chef Software customer support](https://www.chef.io/support/).
+
+#### Cookbook Locations
+
+Upgrading a node means upgrading its cookbooks.
+Speed up the upgrade process by locating cookbooks on your system before you begin.
+Ideally, you can get the cookbooks from their canonical source (that is, `git clone` or another similar version control operation). If you're working with a version control system, you can make and test your changes locally and then push the changes back to the cookbook's source.
+If you can't locate a cookbook, do not download it from an external source, such as the public Chef Supermarket. The cookbook version in your development environment must match the version on your node. [As a last resort](/upgrade_labs/#cookbooks-on-the-chef-server), the Upgrade Lab can get copies of your cookbooks from the Chef Infra Server during the `capture` phase.
+
+Likely cookbook locations:
+
+* Checked into your version control system
+* On a private Supermarket installation
+* In an existing [cookbook development directory](/upgrade_labs/#expected-cookbook-directory-layout)
+
+##### Cookbook Directory Layout
+
+If you have access to cookbook sources, it is simplest if you have the cookbooks stored in one parent directory, similar to this:
+
+```
+/Users/user_name/my_cookbooks/
+  ├── cron
+  │   ├── .git/   # Or other version control bookkeeping
+  │   ├── recipes/
+  │   ├──...
+  │   └── metadata.rb
+  └── chef-client
+  │   ├── .cvs/   # Or other version control bookkeeping
+  │   ├── recipes/
+  │   ├──...
+  │   └── metadata.rb
+  └── my_custom_cookbook
+      ├── .git/   # Or other version control bookkeeping
+      ├── recipes/
+      ├──...
+      └── metadata.rb
+```
+
+The Upgrade Lab works if you have cookbooks in different locations, but it involves more prompting from the command line.
+
+## Upgrade Lab
+
+### Inventory your system with Chef Reports
+
+We recommend starting the upgrade process on a node with a simpler a simple setup, such as one with fewer and simpler cookbooks.
+The `chef report` commands make a survey your nodes and cookbooks. Use the reports to identify a good starting place.
+
+#### chef report nodes
+
+Use `chef report nodes SERVER` command to create a report of the nodes in your system from Ohai data. The command:
```

Comment:
It's `chef report nodes -p PROFILE` - the profile is one of your profiles in your credentials file.